client/api.c

```diff
@@ -770,12 +770,30 @@ TDNFAddCmdLinePackages(
          * to our virtual @cmdline repo */
         pszPkgName = pCmdArgs->ppszCmds[nCmdIndex];
 
+        /* check if it's a package file or URL */
+        if (fnmatch("*.rpm", pszPkgName, 0) != 0) {
```

Comment:
fname.src.rpm matches with this regex
```
#include <stdio.h>
#include <fnmatch.h>

int main()
{
    char pattern[] = "*.rpm";
    char string[] = "file.src.rpm";
    int r;

    r = fnmatch(pattern,string,0);
    if( r==0 )
        printf("'%s' matches '%s'\n",
                pattern,
                string
              );
    else if( r==FNM_NOMATCH )
        printf("'%s' does not match '%s'\n",
                pattern,
                string
              );
    else
        fprintf(stderr,"fnmatch() error\n");

    return(0);
}
````

Reply:
Yes, it does :-). It's supposed to match. If it matches, `fnmatch()` returns `0` and then we check if it also matches `*.src.rpm`.

Reply:
Oops, got it now. I missed the `!=0` part